pinot-core/src/main/java/org/apache/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -201,4 +201,13 @@ private static int getScanBasedFilterPriority(ScanBasedFilterOperator scanBasedF
       return basePriority + 1;
     }
   }
+
+  /**
+   * Perform a composite check to see if there is a FST index present
+   * @param dataSource Datasource to be checked
+   * @return
+   */
+  private static boolean isFSTIndexPresent(DataSource dataSource) {
+    return (dataSource.getFSTIndex() != null || dataSource.getNativeFSTIndex() != null);
```

Comment:
what is the difference between FSTIndex and NativeFSTIndex?

Reply:
FSTIndex is built on Lucene and NativeFSTIndex is built on Pinot's native FST